kahuna/public/js/upload/jobs/upload-jobs.html

```diff
@@ -1,6 +1,6 @@
 <div class="flex-container section-heading current-uploads">
     <h2 class="flex-spacer">Your current uploads <em ng-if="ctrl.remaining">({{ctrl.remaining}} remaining)</em></h2>
-    <a ng-controller="SessionCtrl" class="coloured-link" ui-sref="search.results({uploadedBy: user.email, nonFree: true})"
+    <a ng-controller="SessionCtrl" class="coloured-link" ng-href="/search?uploadedBy={{user.email}}&nonFree=true"
```

Comment:
Hi @andrew-nowak , regarding this change to use `ng-href` instead of `ui-sref` because `ui-sref` actions are not a part of  `$locationChangeStart` events
because it uses the `$location` provider,
and  it can be handled by `$stateChangeStart` since it happens when the state change
so either keep it as it is or revert it back to `ui-sref` and use `$stateChangeStart`
what do you think?
```
$scope.$on('$stateChangeStart', (event, toState, toParams, fromState, fromParams) => {
      if (fromState.url !== toState.url && fromState.url === "/upload") {
        ctrl.displayWarning(event);
    }
  });
```

Reply:
If you find that `$stateChangeStart` works, I think I'd be a lot more comfortable with that. As it happens, it looks like ui-router catches clicks on `<a href="...">`s  and turns them into state transitions - but this seems to be undocumented and definitely took me by surprise, so I'd like to avoid relying on it if at all possible.

That said, I've tried it a few times and haven't directly found any problems with it, so if we can't get `$stateChangeStart` to work then I guess we can go with this approach.

Reply:
Hi Andrew, thanks for the feedback, at this moment this PR has been tested on Edge, Firefox, Chrome, and Safari and working as expected
I will double-check and test with the `$stateChangeStart` events handler and back to you

Reply:
Hi @andrew-nowak, it is working but not stable in some cases, so the current approach is more robust and tested successfully, for all browsers